tests/Unit/ApparentHorizons/Test_Expansion.cpp

```diff
@@ -0,0 +1,118 @@
+  EinsteinSolutions::KerrSchild solution(mass, spin, center);
+  const auto vars = solution.solution(cart_coords, t);
+
+  const auto& spatial_metric =
+      get<gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>>(vars);
+  const auto& deriv_spatial_metric =
+      get<EinsteinSolutions::KerrSchild::deriv_spatial_metric<DataVector>>(
+          vars);
+  const auto inverse_spatial_metric =
+      determinant_and_inverse(spatial_metric).second;
+
+  const auto residual = expansion(
+      db::get<StrahlkorperTags::Rhat<Frame::Inertial>>(box),
+      db::get<StrahlkorperTags::Radius<Frame::Inertial>>(box),
+      db::get<StrahlkorperTags::NormalOneForm<Frame::Inertial>>(box),
+      db::get<StrahlkorperTags::D2xRadius<Frame::Inertial>>(box),
+      inverse_spatial_metric,
+      gr::extrinsic_curvature(
+          get<gr::Tags::Lapse<3, Frame::Inertial, DataVector>>(vars),
+          get<gr::Tags::Shift<3, Frame::Inertial, DataVector>>(vars),
+          get<EinsteinSolutions::KerrSchild::deriv_shift<DataVector>>(vars),
+          spatial_metric,
+          get<gr::Tags::DtSpatialMetric<3, Frame::Inertial, DataVector>>(vars),
+          deriv_spatial_metric),
+      raise_or_lower_first_index(
+          compute_christoffel_first_kind(deriv_spatial_metric),
+          inverse_spatial_metric));
+
+  Approx custom_approx = Approx::custom().epsilon(1.e-12);
+  CHECK_ITERABLE_CUSTOM_APPROX(
+      residual,
+      make_with_value<std::remove_cv_t<decltype(residual)>>(residual, 0.0),
+      custom_approx);
+}
+
+void test_minkowski() {
+  // Make surface of radius 2.
+  Strahlkorper<Frame::Inertial> strahlkorper(10, 10, 2.0, {{0, 0, 0}});
+
+  auto box =
+      db::create<db::AddTags<StrahlkorperTags::items_tags<Frame::Inertial>>,
+                 db::AddComputeItemsTags<
+                     StrahlkorperTags::compute_items_tags<Frame::Inertial>>>(
+          strahlkorper);
+
+  const double t = 0.0;
+  const auto& cart_coords =
+      db::get<StrahlkorperTags::CartesianCoords<Frame::Inertial>>(box);
+
+  EinsteinSolutions::Minkowski<3> solution{};
+
+  const auto deriv_spatial_metric =
+      solution.deriv_spatial_metric(cart_coords, t);
+  const auto inverse_spatial_metric =
+      solution.inverse_spatial_metric(cart_coords, t);
+
+  const auto residual = expansion(
+      db::get<StrahlkorperTags::Rhat<Frame::Inertial>>(box),
+      db::get<StrahlkorperTags::Radius<Frame::Inertial>>(box),
+      db::get<StrahlkorperTags::NormalOneForm<Frame::Inertial>>(box),
+      db::get<StrahlkorperTags::D2xRadius<Frame::Inertial>>(box),
+      inverse_spatial_metric, solution.extrinsic_curvature(cart_coords, t),
+      raise_or_lower_first_index(
+          compute_christoffel_first_kind(deriv_spatial_metric),
+          inverse_spatial_metric));
+
+  Approx custom_approx = Approx::custom().epsilon(1.e-12);
+  CHECK_ITERABLE_CUSTOM_APPROX(
+      residual,
+      make_with_value<std::remove_cv_t<decltype(residual)>>(residual, 1.0),
```

Comment:
again, residual has to be a DataVector, unless you generalize the expansion code